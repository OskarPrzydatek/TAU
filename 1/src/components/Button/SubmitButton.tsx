type SubmitButtonProps = {
  label: string;
  dataTestId?: string;
};

export default function SubmitButton({ label, dataTestId }: SubmitButtonProps) {
  return (
    <button data-testid={dataTestId} type="submit">
      {label}
    </button>
  );
}
