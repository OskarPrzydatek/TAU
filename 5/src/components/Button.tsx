type ButtonProps = {
  label: string;
  dataTestID?: string;
  onClick: () => void;
};

export default function Button({ label, dataTestID, onClick }: ButtonProps) {
  return (
    <button data-testid={dataTestID} onClick={onClick}>
      {label}
    </button>
  );
}
