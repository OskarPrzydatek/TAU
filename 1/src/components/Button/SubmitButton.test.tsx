import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from './SubmitButton';

describe('SumbitButton', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<Button label="Login" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure label renders correctly', () => {
    render(<Button label="Login" dataTestId="button" />);
    expect(screen.getByTestId('button')).toHaveTextContent('Login');
  });
});

export {};
