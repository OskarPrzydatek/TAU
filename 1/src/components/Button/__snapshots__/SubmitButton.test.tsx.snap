// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SumbitButton Component Snapshot 1`] = `
<button
  type="submit"
>
  Login
</button>
`;
